pulser-core/pulser/backend/config.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2023 Pulser Development Team
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines the backend configuration classes."""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Any, Literal
+
+import numpy
+
+
+@dataclass
+class BackendConfig:
+    """The base backend configuration.
+
+    Attributes:
+        backend_options: A dictionary of backend specific options.
+    """
+
+    backend_options: dict[str, Any] = field(default_factory=dict)
+
+
+@dataclass
+class EmulatorConfig(BackendConfig):
+    """The configuration for emulator backends.
```

Comment:
@HGSilveri are these options general for any emulator (including Pasqal cloud-based ones)?

Reply:
That's the idea, so that each emulator evolves towards supporting all these features. For the emulators that are not there yet, I figured I could just raise `NotImplementedError`s  if unsupported configurations are provided.